jobbergate-slurp/slurp/s3_ops.py

```diff
@@ -38,6 +43,61 @@ def build_managers():
         ),
     )
 
+    class NextGenApplicationFiles(ApplicationFiles):
+        @classmethod
+        def file_manager_factory(self, application_id: int) -> FileManager:
+            """
+            Build an application file manager.
+            """
+            return cast(
+                FileManager,
+                file_manager_factory(
+                    id=application_id,
+                    s3_client=s3_client.nextgen,
+                    bucket_name=settings.NEXTGEN_S3_BUCKET_NAME,
+                    work_directory=Path(APPLICATIONS_WORK_DIR),
+                    manager_cls=FileManager,
+                    transformations=IO_TRANSFORMATIONS,
+                ),
+            )
+
+    class NextGenJobScriptFiles(JobScriptFiles):
+        @classmethod
+        def file_manager_factory(self, job_script_id: int) -> FileManager:
+            """
+            Build an application file manager.
+            """
+            return cast(
+                FileManager,
+                file_manager_factory(
+                    id=job_script_id,
+                    s3_client=s3_client.nextgen,
+                    bucket_name=settings.NEXTGEN_S3_BUCKET_NAME,
+                    work_directory=Path(JOBSCRIPTS_WORK_DIR),
+                    manager_cls=FileManager,
+                    transformations=IO_TRANSFORMATIONS,
+                ),
+            )
+
+        @classmethod
+        def get_from_json(cls, input_json: dict):
+            """
+            Get job script files from the legacy json file ``job_script_data_as_string``.
+            """
+            main_filename = "application.sh"
+            main_file_path = Path(main_filename)
+
+            try:
+                unpacked_data = json.loads(input_json)
+                job_script = unpacked_data.get(main_filename, "")
+            except json.JSONDecodeError:
+                job_script = ""
```

Comment:
I wonder if it's better to raise a more meaningful exception here (something more descriptive than JSONDecodeError) and handle that exception in the code that calls this. It feels a bit suspect to eat the exception here and just return an empty string, especially since empty strings are used to indicate that the `main_filename` is missing.